### PR TITLE
Use ResultsModal with Map view instead of plain popup

### PR DESCRIPTION
**Pull request by snyderc**
Clicking on a pin on the map will bring up the ResultsModal instead of a popup within the map.

Usability notes:
- There is no visible way to close the modal; the user needs to know to click outside of the modal to close. Consider adding a visible way to close the modal.
- The modal is tall, particularly on smaller screens (my laptop is 1280x800 as a 13-inch Macbook Pro); the user needs to know to scroll wheel or two-finger touchpad scroll to view the entire modal. Consider making the modal wider, the font smaller, or other way of making the modal a manageable size.

### CODE REVIEW

File: components/subcomponents/Map.js [hunk 1]
```diff
@@ -5,27 +5,8 @@ import GoogleMapReact from 'google-map-react';
...
-    'All eligible people statewide',
-    'Eligible populations statewide'
-];
-
-const doesSiteServeAllEligiblePeopleStatewide = serves => ELIGIBLE_PEOPLE_STATEWIDE_TEXT.includes(serves?.trim());
-
-const isSiteAMassVaccinationSite = locationName => MASS_VACCINATION_SITES.includes(locationName);
-
 const parseLocationData = (data) => {
+    console.log(data);
```
Comment (hblumberg) [inline components/subcomponents/Map.js]: Accidental add? (:

File: utils/site-utils.js [hunk 2]
```diff
@@ -56,6 +61,39 @@ const parseDate = (dateString) => {
...
+        return 'dot';
+    } else if (isSiteAMassVaccinationSite(locationName)) {
+        return 'star star-red';
+    } else if (doesSiteServeAllEligiblePeopleStatewide(serves)) {
+        return 'star star-green';
+    } else {
+        return 'star star-blue';
+    }
+};
+
```
Comment (hblumberg) [inline utils/site-utils.js:96]: Ooh yes, love this re-org!

File: components/subcomponents/Map.js [hunk 3]
```diff
@@ -176,12 +142,6 @@ const Map = ({rawSiteData, center, zoom, onMapChange}) => {
                         getSiteDataByKey={getSiteDataByKey}
                     />
                 ))}
-                {popupData.data && (<Popup
```
Comment (hblumberg) [inline components/subcomponents/Map.js:179]: Can we delete the `Popup` component now?

File: components/ResultsModal.js [hunk 4]
```diff
@@ -27,11 +27,13 @@ const ResultsModal = ({data, hideModal}) => {
                         rawSiteData={[data]}
                         center={{lat: data.latitude, lng: data.longitude}}
                         zoom={12}
+                        setPopupData={() => {}}
+                        onMapChange={() => {}}
```
Comment (hblumberg) [inline components/ResultsModal.js:31]: Ooh I think we can make these optional and add null checks now that the map component is being used by the modal.

File: components/MapAndListView.js [hunk 5]
```diff
@@ -45,11 +47,27 @@ const MapAndListView = (props) => {
...
+                    address: resultsModalData.data.address,
+                    populationsServed: resultsModalData.data.populationsServed,
+                    vaccineAvailability: resultsModalData.data.vaccineAvailability,
+                    lastUpdated: resultsModalData.data.lastUpdated,
+                    bookAppointmentInformation: resultsModalData.data.bookAppointmentInformation,
+                    latitude: resultsModalData.data.latitude,
+                    longitude: resultsModalData.data.longitude,
+                    instructionsAtSite: resultsModalData.data.instructionsAtSite,
+                    sitePinShape: resultsModalData.data.sitePinShape,
+                }}/>}
```
Comment (hblumberg) [inline components/MapAndListView.js]: Could this just be `data={resultsModal.data}`? (If you like the explicit listing of the fields, that also works for me!)

**Review by hblumberg — APPROVED** [review]
So exciting!! I think not having an "close" button might be a ship-stopper (possibility an accessibility issue), but I think this can land to `develop`! (It's currently set to `main`.)

File: components/ResultsModal.js (hunk 4 above)
Comment (snyderc) [reply components/ResultsModal.js:31]: Since we're in maintenance mode now, I'm going to leave this as is in the interest of time

**Pull request merged**